src/Nethermind/Nethermind.Evm/EOF/EvmObjectFormat.cs

```diff
@@ -247,6 +250,52 @@ public bool TryParseEofHeader(ReadOnlySpan<byte> container, [NotNullWhen(true)]
 
         bool ValidateBody(ReadOnlySpan<byte> container, ref EofHeader? header)
         {
+            SectionHeader[]? codeSections = header.Value.CodeSections;
+            (int typeSectionStart, ushort typeSectionSize) = header.Value.TypeSection;
+            if (codeSections.Length == 0 || codeSections.Any(section => section.Size == 0))
+            {
+                if (_loggingEnabled)
+                {
+                    _logger.Trace($"EIP-3540 : CodeSection size must follow a CodeSection, CodeSection length was {codeSections.Length}");
+                }
+
+                header = null;
+                return false;
+            }
+
+            if (codeSections.Length != (typeSectionSize / 4))
+            {
+                if (_loggingEnabled)
+                {
+                    _logger.Trace($"EIP-4750: Code Sections count must match TypeSection count, CodeSection count was {codeSections.Length}, expected {typeSectionSize / 4}");
+                }
+
+                header = null;
+                return false;
+            }
+
+            if (container[typeSectionStart] != 0 && container[typeSectionStart] != 0)
+            {
+                if (_loggingEnabled)
+                {
+                    _logger.Trace($"EIP-4750: first 2 bytes of type section must be 0s");
+                }
+
+                header = null;
+                return false;
+            }
+
+            if (codeSections.Length > 1024)
```

Comment:
1024?

Reply:
1024 is the Maximum number of allowed code sections in EOF bytecode